src/extension.ts

```diff
@@ -43,32 +59,66 @@ export function activate(context: ExtensionContext) {
     context.subscriptions.push(
         commands.registerCommand('crystal.ameba.disable', () => {
             if (!ameba) return;
-            const editor = window.activeTextEditor;
-            if (editor) ameba.clear(editor.document);
+            outputChannel.appendLine('[Disable] Disabling ameba for this session')
+            ameba.clear();
             ameba = null;
         })
     );
 
     workspace.onDidChangeConfiguration(_ => {
         if (!ameba) return;
+        outputChannel.appendLine(`[Config] Reloading diagnostics after config change`)
         ameba.config = getConfig();
+        ameba.clear()
+        executeAmebaOnWorkspace(ameba)
     });
 
-    workspace.textDocuments.forEach(doc => {
-        ameba && ameba.execute(doc);
-    });
+    executeAmebaOnWorkspace(ameba);
 
     workspace.onDidOpenTextDocument(doc => {
         ameba && ameba.execute(doc);
     });
 
     workspace.onDidSaveTextDocument(doc => {
-        if (ameba && ameba.config.onSave) ameba.execute(doc);
+        if (ameba && doc.languageId === 'crystal' && !doc.isUntitled && doc.uri.scheme === 'file') {
+            outputChannel.appendLine(`[Save] Running ameba on ${getRelativePath(doc)}`)
+            ameba.execute(doc);
+        } else if (ameba && path.basename(doc.fileName) == ".ameba.yml") {
+            outputChannel.appendLine(`[Config] Reloading diagnostics after config file change`)
+            ameba.clear();
+            executeAmebaOnWorkspace(ameba);
+        }
```

Comment:
These changes are not in the PR scope.

Reply:
Do you just mean reloading all diagnostics when changing the config?

Reply:
Does that work?